### DISCUSSION TITLE
A-GEM implementation

### DISCUSSION THREAD

**Discussion by JonasFrey96** [ContinualAI/avalanche#448] [Q&A] [answered]
Hello, 

thanks for all your work. Could you explain why the A-GEM numbers are so bad ?
I ran `examples/gem_agem_mnist.py` and the performance on `smnist` are off by a big margin compared to GEM even when increasing the memory capacity. 

Thanks a lot in advance. 

Best regards from Zurich, 
Jonas

**Comment by AndreaCossu** [comment, marked as answer]
Hi @JonasFrey96 :smile: From our experiments, GEM works quite well with single head models, while A-GEM obtains poor performances. With multi-headed models, A-GEM recovers part of the performance in terms of catastrophic forgetting. This is the case also for Synaptic Intelligence: the original paper works with multi head models so the strategy performs poorly with a single head. 
If you have been able to make A-GEM work with single-head , please let us know. That could indicate a bug in our implementation.  However, up to now we have no sign pointing in that direction.

**Reply by JonasFrey96** [reply]
Thanks a lot for you quick response. I will investigate this a little bit further in the upcoming days. I will check the original Tensorflow implementation and let you know if i have more information.

**Reply by AndreaCossu** [reply]
Great, thanks!

**Comment by vlomonaco** [comment]
Moving this issue to "discussions", please @JonasFrey96 mark it as "answered" when you information need is satisfied! :)
